### PR TITLE
Make enforcement of the presence of metic name on pushes optional.

### PR DESCRIPTION
**Pull request by tomwilkie**
Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

**Comment by bboreham** [comment]
Would be nice to have the background to this change.

File: pkg/util/validation/validate.go [hunk 1]
```diff
@@ -65,12 +65,14 @@ func (cfg *Overrides) ValidateSample(userID string, metricName []byte, s client.
 // ValidateLabels returns an err if the labels are invalid.
 func (cfg *Overrides) ValidateLabels(userID string, ls []client.LabelPair) error {
 	metricName, err := extract.MetricNameFromLabelPairs(ls)
-	if err != nil {
-		return httpgrpc.Errorf(http.StatusBadRequest, errMissingMetricName)
-	}
+	if cfg.EnforceMetricName(userID) {
```
Comment (bboreham) [inline pkg/util/validation/validate.go]: In the case where we don't enforce, and it is blank, the error returns below (lines 81 and 105) become less usable.

**Comment by tomwilkie** [comment]
> Would be nice to have the background to this change.

We're using the same label validation code for Loki; but logs don't have metric names.

File: pkg/util/validation/validate.go (hunk 1 above)
Comment (tomwilkie) [reply pkg/util/validation/validate.go]: Yeah, bit of a shame.  Could return the whole label set there - WDYT?

**Pull request merged**